src/witan/app/test_data.clj

```diff
@@ -10,13 +11,37 @@
   (let [;; add users
         user1 (user/add-user! {:name "Mastodon 1" :username "support@mastodonc.com" :password "secret"})
         user2 (user/add-user! {:name "Mastodon 2" :username "support2@mastodonc.com" :password "secret"})
-
         ;; add models
-        m1 (model/add-model! {:name "My Model 1" :description "Description of my model" :owner (:id user1)})
-        m2 (model/add-model! {:name "My Model 2" :description "Description of my model" :owner (:id user2)
-                              :properties [{:name "Some field" :type "text" :context "Placeholder value 123"}]})
-        m3 (model/add-model! {:name "My Model 3" :description "Model with enum" :owner (:id user2)
-                              :properties [{:name "Boroughs" :type "dropdown" :context "Choose a borough" :enum_values ["Camden" "Richmond Upon Thames" "Hackney" "Barnet"]}]})
+        m1 (model/add-model! {:name "My Model 1"
+                              :description "Description of my model"
+                              :owner (:id user1)
+                              :input-data [{:category "Base population data"}]
+                              :output-data [{:category "wishful thinking"}]})
+        m2 (model/add-model! {:name "My Model 2"
+                              :description "Description of my model"
+                              :owner (:id user2)
+                              :properties [{:name "Some field" :type "text" :context "Placeholder value 123"}]
+                              :input-data [{:category "Base population data"}]
+                              :output-data [{:category "All the population data"}]})
+        m3 (model/add-model! {:name "My Model 3"
+                              :description "Model with enum"
+                              :owner (:id user2)
+                              :properties [{:name "Boroughs" :type "dropdown" :context "Choose a borough" :enum_values ["Camden" "Richmond Upon Thames" "Hackney" "Barnet"]}]
+                              :input-data [{:category "long population"}
+                                           {:category "overlay housing"}
+                                           {:category "trend data"}]
+                              :output-data [{:category "housing-linked population"}]})
+        ;; add data
+        d1 (data/add-data! {:category "long population"
+                            :name "London base population"
+                            :publisher (:id user1)
+                            :s3-url "https://s3.eu-central-1.amazonaws.com/witan-test-data/Long+Pop.csv"
+                            :model-id (:model_id m1)})
```

Comment:
Challenging my own understanding again, this data item seems to refer to `m1` and a category of `"long population"`, although `m1` doesn't specify this category so...is that a mistake or do I misunderstand?